patches/api/0333-Added-getHostname-to-AsyncPlayerPreLoginEvent.patch

```diff
@@ -0,0 +1,62 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MCMDEV <john-m.1@gmx.de>
+Date: Fri, 24 Sep 2021 17:59:23 +0200
+Subject: [PATCH] Added getHostname to AsyncPlayerPreLoginEvent
+
+
+diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+index 694a81769076ea58aae9f14f076ab80c9952c957..d30cfb3ef713c026c56f8da19001c4f576d9c355 100644
+--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
++++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+@@ -22,6 +22,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
+     private final InetAddress ipAddress;
+     private final InetAddress rawAddress; // Paper
+     //private UUID uniqueId; // Paper - Not used anymore
++    private final String hostname; // Paper
+ 
+     @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress) {
+@@ -66,7 +67,13 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         this(name, ipAddress, ipAddress, uniqueId, profile);
+     }
+ 
++    // Paper start
++    @Deprecated
+     public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile) {
++        this(name, ipAddress, rawAddress, uniqueId, profile, "");
++    }
++
++    public AsyncPlayerPreLoginEvent(@NotNull final String name, @NotNull final InetAddress ipAddress, @NotNull final InetAddress rawAddress, @NotNull final UUID uniqueId, @NotNull PlayerProfile profile, String hostname) {
+         super(true);
+         this.profile = profile;
+         // Paper end
+@@ -76,7 +83,9 @@ public class AsyncPlayerPreLoginEvent extends Event {
+         this.ipAddress = ipAddress;
+         this.rawAddress = rawAddress; // Paper
+         //this.uniqueId = uniqueId; // Paper - Not used anymore
++        this.hostname = hostname;
+     }
++    // Paper end
```

Comment:
Nice, on reducing this diff here, but now that you have, the Paper comment encompass far more than is actually changing. You can start the paper comment block on line 26 (of this patch) and end on line 29, with two more inline comments for the `@Deprecated` and the `this.hostname = hostname` lines.

Reply:
Should be fixed